### ISSUE TITLE
Field type "structure" is saved as an escaped string

### ISSUE THREAD

**Issue by gdeer81** [caribou/caribou-admin#7]
in the admin interface open a record for editing, edit a field that is a structure with ["foo" "bar" "baz"]
database debug log prints "\"[\\"foo\\"bar\\"\\"baz\\"]\"""

Then if you use this value on your page, lets say you have code to iterate through each value and put it into a <li> tags you get each character listed.

The obvious workaround is to use read-string everywhere you're pulling out these values, except when you encounter values that were imported or stored through some other means that actually are data structures. In those cases you get class cast errors. 

Whether you want to just document this or fix it, I'm happy to help out.

**Comment by quile** [comment]
Hey, thanks for pointing this out.  The "structure" field type got added after most of the admin was written, so it's defaulting to using a "string" type for those fields.  I'll try to take a look and see if I can fix it; there is actually a "structure" editor (although it's a bit rough) that lets you build up complex structures but it was built specifically for siphons so might not generalise to structures quite as easily as I'd want.  

You _definitely_ shouldn't have to use read-string, because the system is already calling read-string for you!  It's something worth fixing, if we can...

**Comment by quile** [comment]
Yep, looks like I stubbed out a "StructureFieldEditor" two years ago but it just inherits all its behaviour from text fields; this is where the fix will have to go:

https://github.com/caribou/caribou-admin/blob/master/resources/public/_admin/js/modules/pages/content/editors/structure.js

It's on my list now; hopefully I'll be able to fix it soon and push out a new release.
